.github/workflows/terraform.yml

```diff
@@ -0,0 +1,48 @@
+name: "terraform"
+
+# for security should only run on push to bors branches (staging/trying)
+on:
+  push:
+    branches:
+      - staging
+      - trying
+
+jobs:
+  terraform-deploy:
+    if: github.repository == 'nix-community/infra'
+    runs-on: ubuntu-latest
+    env:
+      # TODO: add to repo secrets
+      SOPS_AGE_KEY: ${{ secrets.SOPS_AGE_KEY }}
+      TF_IN_AUTOMATION: true
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dorny/paths-filter@v2
```

Comment:
I would have preferred not to depend on another action but handling the diff/filter logic ourselves would be a maintenance burden. We could run it unconditionally on every `bors merge` but then any terraform problems would block non-terraform PRs.